homeassistant/components/mqtt/__init__.py

```diff
@@ -287,20 +287,31 @@ async def finish_dump(_: datetime) -> None:
 
 async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Migrate the options from config entry data."""
+    _LOGGER.debug("Migrating from version %s:%s", entry.version, entry.minor_version)
     data: dict[str, Any] = dict(entry.data)
     options: dict[str, Any] = dict(entry.options)
-    for key in ENTRY_OPTION_FIELDS:
-        if key not in data:
-            continue
-        options[key] = data.pop(key)
-    hass.config_entries.async_update_entry(
-        entry,
-        data=data,
-        options=options,
-        version=ENTRY_VERSION,
-        minor_version=ENTRY_MINOR_VERSION,
+    if entry.version > 1:
+        # This means the user has downgraded from a future version
+        return False
+
+    if entry.version == 1 and entry.minor_version < 2:
+        # Can be removed when config entry is bumped to version 2.1
+        # with HA Core 2026.1.0
```

Comment:
Just to be clear, we should introduce *read support* for version 2.1 earlier than 2026.1, then we write version 2.1 in 2026.1.